frontend/src/embedding-sdk-bundle/components/public/dashboard/use-common-dashboard-params.tsx

```diff
@@ -127,19 +131,23 @@ const findDashboardById = (
   dashboardId: DashboardId,
   dashboards: Record<DashboardId, StoreDashboard>,
 ): StoreDashboard | null => {
-  if (typeof dashboardId === "number" || isJWT(dashboardId)) {
-    return dashboards[dashboardId] ?? null;
-  }
-
   // Lookup via entity ids.
   // Dashboards are a mapping of numeric id to dashboard.
-  if (typeof dashboardId === "string") {
+  if (isBaseEntityID(dashboardId)) {
```

Comment:
We need to check the entity ID before, because it's possible that we can have a numeric ID string with the same length as the entity ID that is an actual entity ID, although realistically, the probability is probably lower than winning a lottery.